`testCached` doesn't work for `ScalaJSModule`.

```
java.lang.ExceptionInInitializerError
java.lang.ExceptionInInitializerError
        at org.scalatest.tools.TaskRunner.executionFuture(TaskRunner.scala:80)
        at org.scalatest.tools.TaskRunner.execute(TaskRunner.scala:74)
        at mill.scalalib.TestRunner$.$anonfun$runTests$2(TestRunner.scala:102)
        at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
        at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
        at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
        at scala.collection.TraversableLike.map(TraversableLike.scala:238)
        at scala.collection.TraversableLike.map$(TraversableLike.scala:231)
        at scala.collection.AbstractTraversable.map(Traversable.scala:108)
        at mill.scalalib.TestRunner$.$anonfun$runTests$1(TestRunner.scala:81)
        at mill.modules.Jvm$.inprocess(Jvm.scala:184)
        at mill.scalalib.TestRunner$.runTests(TestRunner.scala:76)
        at mill.scalalib.TestRunner$.main(TestRunner.scala:52)
        at mill.scalalib.TestRunner.main(TestRunner.scala)
Caused by: java.lang.ClassCastException: scala.scalajs.js.Object cannot be cast to scala.scalajs.js.Dictionary
        at scala.scalajs.js.Dictionary$.empty(Dictionary.scala:83)
        at scala.scalajs.reflect.Reflect$.<init>(Reflect.scala:72)
        at scala.scalajs.reflect.Reflect$.<clinit>(Reflect.scala)
        ... 15 more
1 targets failed
```

--- a/scalajslib/src/ScalaJSModule.scala
+++ b/scalajslib/src/ScalaJSModule.scala
@@ -203,7 +203,7 @@ trait TestScalaJSModule extends ScalaJSModule with TestModule {
 
   override def testLocal(args: String*) = T.command { test(args:_*) }
 
-  override def testTask(args: Task[Seq[String]]): Task[(String, Seq[TestRunner.Result])] = T.task {
+  override protected def testTask(args: Task[Seq[String]]): Task[(String, Seq[TestRunner.Result])] = T.task {
     val (close, framework) = mill.scalajslib.ScalaJSWorkerApi.scalaJSWorker().getFramework(
       toolsClasspath().map(_.path),
       jsEnvConfig(),

--- a/scalajslib/src/ScalaJSModule.scala
+++ b/scalajslib/src/ScalaJSModule.scala
@@ -8,6 +8,7 @@ import mill.scalalib.Lib.resolveDependencies
 import mill.scalalib.{DepSyntax, Lib, TestModule, TestRunner}
 import mill.util.Ctx
 import mill.api.Loose
+import mill.define.Task
 import mill.scalajslib.api._
 trait ScalaJSModule extends scalalib.ScalaModule { outer =>
 
@@ -202,22 +203,22 @@ trait TestScalaJSModule extends ScalaJSModule with TestModule {
 
   override def testLocal(args: String*) = T.command { test(args:_*) }
 
-  override def test(args: String*) = T.command {
+  override def testTask(args: Task[Seq[String]]): Task[(String, Seq[TestRunner.Result])] = T.task {
     val (close, framework) = mill.scalajslib.ScalaJSWorkerApi.scalaJSWorker().getFramework(
-        toolsClasspath().map(_.path),
-        jsEnvConfig(),
-        testFrameworks().head,
-        fastOptTest().path.toIO,
-        moduleKind()
-      )
+      toolsClasspath().map(_.path),
+      jsEnvConfig(),
+      testFrameworks().head,
+      fastOptTest().path.toIO,
+      moduleKind()
+    )
 
     val (doneMsg, results) = TestRunner.runTests(
-        _ => Seq(framework),
-        runClasspath().map(_.path),
-        Agg(compile().classes.path),
-        args,
-        T.testReporter
-      )
+      _ => Seq(framework),
+      runClasspath().map(_.path),
+      Agg(compile().classes.path),
+      args(),
+      T.testReporter
+    )
     val res = TestModule.handleResults(doneMsg, results)
     // Hack to try and let the Node.js subprocess finish streaming it's stdout
     // to the JVM. Without this, the stdout can still be streaming when `close()`

--- a/scalajslib/test/src/HelloJSWorldTests.scala
+++ b/scalajslib/test/src/HelloJSWorldTests.scala
@@ -150,67 +150,96 @@ object HelloJSWorldTests extends TestSuite {
       'artifactId_100RC1 - testArtifactId("2.12.4", "1.0.0-RC1", "hello-js-world_sjs1.0-RC1_2.12")
       'artifactId_100 - testArtifactId("2.12.4", "1.0.0", "hello-js-world_sjs1_2.12")
     }
+
+    def runTests(testTask: define.NamedTask[(String, Seq[TestRunner.Result])]): Map[String, Map[String, TestRunner.Result]] = {
+      val Left(Result.Failure(_, Some(res))) = helloWorldEvaluator(testTask)
+
+      val (doneMsg, testResults) = res
+      testResults
+        .groupBy(_.fullyQualifiedName)
+        .mapValues(_.map(e => e.selector -> e).toMap)
+    }
+
+    def checkUtest(scalaVersion: String, scalaJSVersion: String, cached: Boolean) = {
+      val resultMap = runTests(
+        if(!cached) HelloJSWorld.buildUTest(scalaVersion, scalaJSVersion).test.test()
+        else HelloJSWorld.buildUTest(scalaVersion, scalaJSVersion).test.testCached
+      )
+
+      val mainTests = resultMap("MainTests")
+      val argParserTests = resultMap("ArgsParserTests")
+
+      assert(
+        mainTests.size == 2,
+        mainTests("MainTests.vmName.containJs").status == "Success",
+        mainTests("MainTests.vmName.containScala").status == "Success",
+
+        argParserTests.size == 2,
+        argParserTests("ArgsParserTests.one").status == "Success",
+        argParserTests("ArgsParserTests.two").status == "Failure"
+      )
+    }
+
+    def checkScalaTest(scalaVersion: String, scalaJSVersion: String, cached: Boolean) = {
+      val resultMap = runTests(
+        if(!cached) HelloJSWorld.buildScalaTest(scalaVersion, scalaJSVersion).test.test()
+        else HelloJSWorld.buildScalaTest(scalaVersion, scalaJSVersion).test.testCached
+      )
+
+      val mainSpec = resultMap("MainSpec")
+      val argParserSpec = resultMap("ArgsParserSpec")
+
+      assert(
+        mainSpec.size == 2,
+        mainSpec("vmName should contain js").status == "Success",
+        mainSpec("vmName should contain Scala").status == "Success",
+
+        argParserSpec.size == 2,
+        argParserSpec("parse should one").status == "Success",
+        argParserSpec("parse should two").status == "Failure"
+      )
+    }
+
     'test - {
-      def runTests(testTask: define.Command[(String, Seq[TestRunner.Result])]): Map[String, Map[String, TestRunner.Result]] = {
-        val Left(Result.Failure(_, Some(res))) = helloWorldEvaluator(testTask)
-
-        val (doneMsg, testResults) = res
-        testResults
-          .groupBy(_.fullyQualifiedName)
-          .mapValues(_.map(e => e.selector -> e).toMap)
-      }
-
-      def checkUtest(scalaVersion: String, scalaJSVersion: String) = {
-        val resultMap = runTests(HelloJSWorld.buildUTest(scalaVersion, scalaJSVersion).test.test())
-
-        val mainTests = resultMap("MainTests")
-        val argParserTests = resultMap("ArgsParserTests")
-
-        assert(
-          mainTests.size == 2,
-          mainTests("MainTests.vmName.containJs").status == "Success",
-          mainTests("MainTests.vmName.containScala").status == "Success",
-
-          argParserTests.size == 2,
-          argParserTests("ArgsParserTests.one").status == "Success",
-          argParserTests("ArgsParserTests.two").status == "Failure"
-        )
-      }
-
-      def checkScalaTest(scalaVersion: String, scalaJSVersion: String) = {
-        val resultMap = runTests(HelloJSWorld.buildScalaTest(scalaVersion, scalaJSVersion).test.test())
-
-        val mainSpec = resultMap("MainSpec")
-        val argParserSpec = resultMap("ArgsParserSpec")
-
-        assert(
-          mainSpec.size == 2,
-          mainSpec("vmName should contain js").status == "Success",
-          mainSpec("vmName should contain Scala").status == "Success",
-
-          argParserSpec.size == 2,
-          argParserSpec("parse should one").status == "Success",
-          argParserSpec("parse should two").status == "Failure"
-        )
-      }
-
-      'utest_21112_0622 - TestUtil.disableInJava9OrAbove(checkUtest("2.11.12", "0.6.22"))
-      'utest_2124_0622 - checkUtest("2.12.4", "0.6.22")
-      'utest_21112_0631 - TestUtil.disableInJava9OrAbove(checkUtest("2.11.12", "0.6.31"))
-      'utest_2124_0631 - checkUtest("2.12.4", "0.6.31")
+      val cached = false
+      'utest_21112_0622 - TestUtil.disableInJava9OrAbove(checkUtest("2.11.12", "0.6.22", cached))
+      'utest_2124_0622 - checkUtest("2.12.4", "0.6.22", cached)
+      'utest_21112_0631 - TestUtil.disableInJava9OrAbove(checkUtest("2.11.12", "0.6.31", cached))
+      'utest_2124_0631 - checkUtest("2.12.4", "0.6.31", cached)
 //      No utest artifact for Scala.js 1.0.0 published yet
 //      'utest_21112_100 - TestUtil.disableInJava9OrAbove(checkUtest("2.11.12", "1.0.0"))
 //      'utest_2124_100 - checkUtest("2.12.4", "1.0.0")
 
       // No test for ScalaTest with 0.6.22 because ScalaTest 3.1.0 requires Scala.js 0.6.29+
-      'scalaTest_21112_0631 - TestUtil.disableInJava9OrAbove(checkScalaTest("2.11.12", "0.6.31"))
-      'scalaTest_2124_0631 - checkScalaTest("2.12.4", "0.6.31")
-      'scalaTest_21112_100RC1 - checkScalaTest("2.11.12", "1.0.0-RC1")
-      'scalaTest_2124_100RC1 - checkScalaTest("2.12.4", "1.0.0-RC1")
+      'scalaTest_21112_0631 - TestUtil.disableInJava9OrAbove(checkScalaTest("2.11.12", "0.6.31", cached))
+      'scalaTest_2124_0631 - checkScalaTest("2.12.4", "0.6.31", cached)
+      'scalaTest_21112_100RC1 - checkScalaTest("2.11.12", "1.0.0-RC1", cached)
+      'scalaTest_2124_100RC1 - checkScalaTest("2.12.4", "1.0.0-RC1", cached)
 //      No ScalaTest artifact for Scala.js 1.0.0 published yet
 //      'scalaTest_21112_100 - checkScalaTest("2.11.12", "1.0.0")
 //      'scalaTest_2124_100 - checkScalaTest("2.12.4", "1.0.0")
     }
+
+    'testCached - {
+      val cached = true
+      'utest_21112_0622 - TestUtil.disableInJava9OrAbove(checkUtest("2.11.12", "0.6.22", cached))
+      'utest_2124_0622 - checkUtest("2.12.4", "0.6.22", cached)
+      'utest_21112_0631 - TestUtil.disableInJava9OrAbove(checkUtest("2.11.12", "0.6.31", cached))
+      'utest_2124_0631 - checkUtest("2.12.4", "0.6.31", cached)
+      //      No utest artifact for Scala.js 1.0.0 published yet
+      //      'utest_21112_100 - TestUtil.disableInJava9OrAbove(checkUtest("2.11.12", "1.0.0"))
+      //      'utest_2124_100 - checkUtest("2.12.4", "1.0.0")
+
+      // No test for ScalaTest with 0.6.22 because ScalaTest 3.1.0 requires Scala.js 0.6.29+
+      'scalaTest_21112_0631 - TestUtil.disableInJava9OrAbove(checkScalaTest("2.11.12", "0.6.31", cached))
+      'scalaTest_2124_0631 - checkScalaTest("2.12.4", "0.6.31", cached)
+      'scalaTest_21112_100RC1 - checkScalaTest("2.11.12", "1.0.0-RC1", cached)
+      'scalaTest_2124_100RC1 - checkScalaTest("2.12.4", "1.0.0-RC1", cached)
+      //      No ScalaTest artifact for Scala.js 1.0.0 published yet
+      //      'scalaTest_21112_100 - checkScalaTest("2.11.12", "1.0.0")
+      //      'scalaTest_2124_100 - checkScalaTest("2.12.4", "1.0.0")
+    }
+
 
     def checkRun(scalaVersion: String, scalaJSVersion: String): Unit = {
       val task = HelloJSWorld.helloJsWorld(scalaVersion, scalaJSVersion).run()

